src/mongocxx/collection.cpp

```diff
@@ -367,7 +371,8 @@ bsoncxx::stdx::optional<bsoncxx::document::value> collection::find_one_and_updat
         rd == options::return_document::k_after, reply.bson(), &error);
 
     if (!r) {
-        throw exception::write(std::move(_impl->gle()), std::make_tuple(error.message, error.code));
+        throw exception::write{std::move(_impl->gle()),
+                               exception::error_code_from_bson_error_t(error), error.message};
```

Comment:
the problem is that the constructor would have to go in a public header and we don't want to expose bson_error_t to users.